compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -884,16 +884,25 @@ trait Implicits { self: Typer =>
    *
    *     <parent> {
    *       MirroredMonoType = <monoType>
-   *       MirroredTypeConstrictor = <tycon>
-   *       MirroredLabel = <label> }
+   *       MirroredType = <mirroredType>
+   *       MirroredLabel = <label>
+   *       Scope = <scope> // if scope ne NoType
+   *     }
    */
-  private def mirrorCore(parent: Type, monoType: Type, mirroredType: Type, label: Name)(implicit ctx: Context) = {
-    parent
+  private def mirrorCore(parent: Type, monoType: Type, mirroredType: Type, label: Name, scope: Type)(implicit ctx: Context) = {
+    val m = parent
       .refinedWith(tpnme.MirroredMonoType, TypeAlias(monoType))
       .refinedWith(tpnme.MirroredType, TypeAlias(mirroredType))
       .refinedWith(tpnme.MirroredLabel, TypeAlias(ConstantType(Constant(label.toString))))
+    if (scope eq NoType) m else m.refinedWith(tpnme.Scope, TypeAlias(scope))
   }
 
+  private def scope(formal: Type)(implicit ctx: Context): Type =
+    formal.member(tpnme.Scope).info match {
```

Comment:
That seems a bit ad-hoc, why not bring along all type members in `formal` instead as a generalization ?

Reply:
Yes, that would work.